ambry-api/src/main/java/com/github/ambry/rest/RequestPath.java

```diff
@@ -130,6 +130,8 @@ public static RequestPath parse(String path, Map<String, Object> args, List<Stri
           throw new RestServiceException("Segment index given is not an integer", RestServiceErrorCode.BadRequest);
         }
         //otherwise, nothing to do
+      } catch (ArrayIndexOutOfBoundsException e) {
+        throw new RestServiceException("Bad request path", RestServiceErrorCode.BadRequest);
```

Comment:
let's not do this, let's find the code that use out of bound index and check the length of the array,  throw a RestServiceException.

Reply:
The code that used out-of-bound index is L120
```
if (SEGMENT.equals(fields[fields.length - 2])) {
```

What's the difference between the approach in this PR and the one you are suggesting ? The end result in either case is a RestServiceException.

Reply:
Let's lost the root cause exception.  Normally, we if a wrap an other exception, we capture it as the cause.  I notice the other new RestServiceException() doesn't capture the cause exception.  It's a good practice to capture the cause exception.

Reply:
good point !

Reply:
`ArrayIndexOutOfBoundsException` is a RuntimeException. It's a bit like `NullPointerException`. If you already know which logic would yield this exception, then we rely on if-else statement, not try-catch statement. It's faster with if-else.